Sources/LeafKit/LeafLexer.swift

```diff
@@ -79,49 +109,28 @@ struct TemplateSource {
     }
 }
 
-public struct LexerError: Error {
-    public enum Reason {
-        case invalidTagToken(Character)
-        case invalidParameterToken(Character)
-        case unterminatedStringLiteral
-    }
-    
-    public let line: Int
-    public let column: Int
-    public let name: String
-    public let reason: Reason
-    public let lexed: [LeafToken]
-    
-    internal init(src: TemplateSource, lexed: [LeafToken], reason: Reason) {
-        self.line = src.line
-        self.column = src.column
-        self.reason = reason
-        self.lexed = lexed
-        self.name = src.name
-    }
-}
-
 struct LeafLexer {
     private enum State {
         // parses as raw, until it finds `#` (excluding escaped `\#`)
-        case normal
+        case raw
         // found a `#`
         case tag
         // found a `(` continues until `)`
-        case parameters(depth: Int)
+        case parameters
```

Comment:
Parameter depth is moved from the lex state to the overall state for the sake of readily accessing it